src/librustc/ty/mod.rs

```diff
@@ -1371,6 +1371,12 @@ pub struct AdtDef {
     pub variants: Vec<VariantDef>,
     flags: AdtFlags,
     pub repr: ReprOptions,
+    /// Indicates whether this abstract data type will be expanded on in future (new
+    /// fields/variants) and as such, whether downstream crates must match exhaustively on the
+    /// fields/variants of this data type.
+    ///
+    /// See RFC 2008 (https://github.com/rust-lang/rfcs/pull/2008).
+    pub non_exhaustive: bool,
```

Comment:
This is better merged into the `flags: AdtFlags` field.

Reply:
Fixed.